components/script/dom/idbobjectstore.rs

```diff
@@ -385,6 +385,26 @@ impl IDBObjectStore {
         self.key_path.is_some()
     }
 
+    /// Checks if the transation is active, throwing a "TransactionInactiveError" DOMException if not.
+    /// If writing is true, it then checks if the transaction is a read-only transaction, throwing a "ReadOnlyError" DOMException if so.
+    fn check_transaction(&self, writing: bool) -> Fallible<()> {
```

Comment:
Instead of using a boolean argument, can we do this:
```rust
fn verify_transaction_active() -> Fallible<()> { ... }
fn verify_readwrite_transaction_active() -> Fallible<()> {
    self.verify_transaction_active()?;
    if let IDBTransactionMode::Readonly = ... { ... }
    Ok(())
}
```

I think that will be easier to read in callers.